rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -524,6 +538,25 @@ impl<B> Inner<B> {
         Self { body }
     }
 
+    async fn next(&mut self) -> Option<Result<Bytes, B::Error>>
+    where
+        Self: Unpin,
```

Comment:
this `Unpin` bound isn't new right? I guess this is on a internal type so it's fine.